octobot_trading/orders/types/limit/stop_loss_limit_order.py

```diff
@@ -25,14 +25,14 @@ def __init__(self, trader, side=TradeOrderSide.SELL):
         self.trigger_above = False
         self.limit_price = self.UNINITIALIZED_LIMIT_PRICE
 
-    async def on_fill(self):
-        await self.trader.create_artificial_order(TraderOrderType.SELL_MARKET
-                                                  if self.side is TradeOrderSide.SELL
-                                                  else TraderOrderType.BUY_MARKET,
-                                                  self.symbol, self.origin_stop_price,
-                                                  self.origin_quantity,
-                                                  self.limit_price
-                                                  if self.limit_price != self.UNINITIALIZED_LIMIT_PRICE else
-                                                  self.origin_stop_price,
-                                                  self.linked_portfolio)
-        await super().on_fill()
+    async def on_trade_creation(self):
```

Comment:
this is to prevent secondary order creation on a cleared order ?

Reply:
Exactly it's called just after portfolio update from the previous Order